### PR TITLE
[WIP] ShimLoader.updateSparkClassLoader fails with openjdk Java11

### PR DESCRIPTION
**Pull request by ekrivokonmapr**
This PR is opened to fix one of the JDK11 support issue: https://github.com/NVIDIA/spark-rapids/issues/3851

java.lang.NoSuchMethodException: No such method: addURL() on object: jdk.internal.loader.ClassLoaders$AppClassLoader
        at org.apache.commons.lang3.reflect.MethodUtils.invokeMethod(MethodUtils.java:231)
        at org.apache.commons.lang3.reflect.MethodUtils.invokeMethod(MethodUtils.java:184)
        at com.nvidia.spark.rapids.ShimLoader$.$anonfun$updateSparkClassLoader$3(ShimLoader.scala:216)
        
Also, deprecated new Integer(_) was replaced.

### CODE REVIEW

**Comment by jlowe** [comment]
Thanks for the contribution, @ekrivokonmapr!  Couple of high-level things that need to be addressed:
- The PR needs to target the default branch, `branch-22.06`, as that is where we are actively developing for the next release,  22.06.
- The PR needs a signoff on at least one commit per the [contributing guide](https://github.com/NVIDIA/spark-rapids/blob/branch-22.06/CONTRIBUTING.md#sign-your-work)

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala [hunk 1]
```diff
@@ -209,24 +209,18 @@ object ShimLoader extends Logging {
   private def updateSparkClassLoader(): Unit = {
     // TODO propose a proper addClassPathURL API to Spark similar to addJar but
     //  accepting non-file-based URI
-    serializerClassloader().foreach { urlAddable =>
```
Comment (gerashegalov) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala:212]: This will break existing deployments for which this logic was critical. 

Yes it's not supported on vanilla JDK11 JVM. We can look at this deeper, but In the past experience `spark.rapids.force.caller.classloader=true` was really necessary only for advanced REPLs such as some multi-session notebook servers. We proposed it as the default value because it was safe on JDK8 and covered all the cases we have encountered. 

I would suggest to check if setting `spark.rapids.force.caller.classloader=false` works for your use cases https://github.com/NVIDIA/spark-rapids/issues/3851#issuecomment-992235387. 

If yes we should set it in code for JDK versions 11+.

File: tests/src/test/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleTestHelper.scala [hunk 2]
```diff
@@ -165,7 +165,7 @@ object RapidsShuffleTestHelper extends MockitoSugar with Arm {
   }
 
   def withMockContiguousTable[T](numRows: Long)(body: ContiguousTable => T): T = {
-    val rows: Seq[Integer] = (0 until numRows.toInt).map(new Integer(_))
+    val rows: Seq[Integer] = (0 until numRows.toInt).map(Integer.valueOf)
```
Comment (gerashegalov) [inline tests/src/test/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleTestHelper.scala:168]: This should be a separate PR because it's orthogonal #3851. It's a good change but it is already fixed by #4338

**Review by gerashegalov — CHANGES_REQUESTED** [review]

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala (hunk 1 above)
Comment (ekrivokonmapr) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala:212]: Thank you, will check it using workaround.

File: tests/src/test/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleTestHelper.scala (hunk 2 above)
Comment (ekrivokonmapr) [reply tests/src/test/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleTestHelper.scala:168]: Thanks for the review.

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala (hunk 1 above)
Comment (ekrivokonmapr) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala:212]: Workaround works for me, will close this PR.
